src/app/pages/course-list/course-list.component.ts

```diff
@@ -13,10 +13,16 @@ export class CourseListComponent{
 
   constructor(private _service : CourseService){}
 
-  onItemDelete(item: Course){
+  onDelete(item: Course){
     var result: boolean = confirm("Do you really want to delete this course? ");
     if (result) {
-      this._service.removeItem(item);
+      this._service.removeItem(item).subscribe(res => {
```

Comment:
it's bad to subscribe to an observable than never ends. you introduce memory leaks by doing this.
it's better to call removeItem() and treat this as an effect.
the service should be responsible for emitting a new updated array of courses in the main stream.